Content.Shared/Power/Generator/ActiveGeneratorRevvingSystem.cs

```diff
@@ -0,0 +1,85 @@
+using Content.Shared.DoAfter;
+
+namespace Content.Shared.Power.Generator;
+
+public sealed class ActiveGeneratorRevvingSystem: EntitySystem
+{
+    [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
+    [Dependency] private readonly EntityManager _entity = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ActiveGeneratorRevvingComponent, AnchorStateChangedEvent>(OnAnchorStateChanged);
+    }
+
+    /// <summary>
+    /// Handles the AnchorStateChangedEvent to stop auto-revving when unanchored.
+    /// </summary>
+    private void OnAnchorStateChanged(EntityUid uid, ActiveGeneratorRevvingComponent component, AnchorStateChangedEvent args)
+    {
+        if (!args.Anchored)
+            StopAutoRevving(uid);
+    }
+
+    /// <summary>
+    /// Start revving a generator entity automatically, without another entity doing a do-after.
+    /// Used for remotely activating a generator.
+    /// </summary>
+    /// <param name="uid">Uid of the generator entity.</param>
+    public void StartAutoRevving(EntityUid uid)
+    {
+        if (TryComp<ActiveGeneratorRevvingComponent>(uid, out var component))
+        {
+            // reset the revving
+            component.CurrentTime = TimeSpan.FromSeconds(0);
+            return;
+        }
```

Comment:
It's better to use resolve for public entity system methods instead of calling `TryComp`
https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html#public-api-method-signature